### PR TITLE
[5.5] Add where whereWeekOfYear in query builder

### PR DESCRIPTION
**Pull request by maldechavda**
add a WEEKOFYEAR in query builder, user like

``` 
$users = User::whereWeek('created_at', 32)
    ->get()
 ```

or 
```
$users = DB::table('users')
    ->whereWeek('created_at', 32)
    ->get();
```

### CODE REVIEW

**Comment by Modelizer** [comment]
It's quite a confusing by looking to syntax that it belongs to a week of year `whereWeek` what if I want to use it as a week of a month then?

**Comment by maldechavda** [comment]
@Modelizer Yes, renamed method name.

**Comment by Modelizer** [comment]
@maldechavda Thanks! 

What if I want to get by month? I know you will create a new method `whereWeekOfMonth` but let @taylorotwell view it first. Also, update the description to get more clarity.

Thanks.

File: src/Illuminate/Database/Query/Grammars/Grammar.php [hunk 1]
```diff
@@ -538,12 +550,12 @@ protected function compileOrders(Builder $query, $orders)
         }
 
         return 'order by '.implode(', ', array_map(function ($order) {
-            if (isset($order['sql'])) {
-                return $order['sql'];
-            }
+                if (isset($order['sql'])) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Grammars/Grammar.php]: why has all this changed?

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1759,7 +1777,7 @@ protected function clearSelectAliases(array $columns)
     {
         return array_map(function ($column) {
             return is_string($column) && ($aliasPosition = strpos(strtolower($column), ' as ')) !== false
-                    ? substr($column, 0, $aliasPosition) : $column;
+                ? substr($column, 0, $aliasPosition) : $column;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: please don't make other unrelated changes to the file

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by maldechavda** [comment]
@GrahamCampbell Thank you, removed unneeded changes.

**Comment by GrahamCampbell** [comment]
Looks like you've actually targeted 5.5, rather than 5.4. Please can you target the 5.4 branch. :P

**Comment by maldechavda** [comment]
@GrahamCampbell So sorry

**Comment by GrahamCampbell** [comment]
Don't worry about it. :)
